zebra/zebra_ptm.c

```diff
@@ -642,6 +642,7 @@ int zebra_ptm_sock_read(struct thread *thread)
 	do {
 		rc = ptm_lib_process_msg(ptm_hdl, sock, ptm_cb.in_data,
 					 ZEBRA_PTM_MAX_SOCKBUF, NULL);
+		memset(ptm_cb.in_data, 0, ZEBRA_PTM_MAX_SOCKBUF);
```

Comment:
is the bug maybe here in `lib/ptm_lib.c` ? it looks like we know the valid message length, in `len`, but we don't use it - we offer the max buffer size in `csv_init()` instead

```
		/* read the rest of the msg */
		len = _ptm_lib_read_ptm_socket(fd, inbuf, msglen);
		if (len <= 0) {
			return (len);
		}

		inbuf[len] = '\0';

		csv = csv_init(NULL, NULL, PTMLIB_MSG_SZ);
```